### PR TITLE
PLATFORM-908 mysqli DB driver for MediaWiki

### PR DESCRIPTION
**Pull request by macbre**
As a part of the upgrade to PHP v5.6 we want to switch to `mysqli` extension to talk to our database (`mysql` extension is no longer maintained and became deprecated). MW 1.22 provides the support for both `mysql` & `mysqli` extensions. This PR is a set of cherry picks from [MediaWiki repository](https://github.com/wikimedia/mediawiki/tree/master/includes/db).

@Wikia/platform-team

### CODE REVIEW

**Comment by michalroszka** [comment]
Looks fine.

File: includes/db/Database.php [hunk 1]
```diff
@@ -585,18 +585,37 @@ public function getSchemaPath() {
...
 
+		if ( is_array( $params ) ) { // MW 1.22
+			$server = $params['host'];
+			$user = $params['user'];
+			$password = $params['password'];
+			$dbName = $params['dbname'];
+			$flags = $params['flags'];
+			$tablePrefix = $params['tablePrefix'];
+		} else { // legacy calling pattern
+			wfDeprecated( __METHOD__ . " method called without parameter array.", "1.23" );
```
Comment (wladekb) [inline includes/db/Database.php]: do we want to spam ourselves here?

Comment (macbre) [inline includes/db/Database.php]: Given the changes in `Database::factory` the `else` block should not be executed.

**Comment by wladekb** [comment]
lgtm

**Comment by macbre** [comment]
Ported more changes from MW repo - 5d58274

**Comment by michalroszka** [comment]
:+1:

**Pull request merged**